pydicom/filereader.py

```diff
@@ -282,6 +294,12 @@ def data_element_generator(fp,
         if length != 0xFFFFFFFF:
             # don't defer loading of Specific Character Set value as it is
             # needed immediately to get the character encoding for other tags
+            if tag_set is not None and tag not in tag_set and tag != (
```

Comment:
A good starting point for finding test files is the [readme.txt](https://github.com/pydicom/pydicom/blob/master/pydicom/tests/test_files/README.txt) file describing them.  Not sure if all are listed in there, or are up to date.  However, the readme says priv_SQ.dcm has an undefined sequence.